pkg/client/restclient/request_test.go

```diff
@@ -1543,7 +1541,7 @@ func testRESTClient(t testing.TB, srv *httptest.Server) *RESTClient {
 		}
 	}
 	versionedAPIPath := testapi.Default.ResourcePath("", "", "")
-	client, err := NewRESTClient(baseURL, versionedAPIPath, defaultContentConfig(), 0, 0, nil, nil)
+	client, err := NewRESTClient([]*url.URL{baseURL}, versionedAPIPath, defaultContentConfig(), 0, 0, nil, nil)
```

Comment:
Please add a new test to request_test.go that verifies that retries properly walk through other alternate hosts and terminate on errors, etc (we really need 100% coverage of request.go, since it's the core of the whole project)